src/ua.c

```diff
@@ -2020,6 +2112,69 @@ struct ua *uag_find(const struct pl *cuser)
 }
 
 
+/**
+ * Find the correct UA from SIP message
+ *
+ * @param msg SIP message
+ *
+ * @return Matching UA if found, NULL if not found
+ */
+struct ua *uag_find_msg(const struct sip_msg *msg)
+{
+	struct le *le;
+	const struct pl *cuser = &msg->uri.user;
```

Comment:
please note that this is a global function, and input parameters should be checked:

```
if (!msg)
    return NULL;
```

if msg is NULL then the program will crash